### PR TITLE
Testing/teacher

### PR DESCRIPTION
**Pull request by KyleWong2510**
## Pull request checklist
Please check if your PR fulfills the following requirements:
- [ ] Lint (`npm run lint`) has passed locally and any fixes were made for failures
## Pull request type
 
 
Please check the type of change your PR introduces:
- [ ] Bugfix
- [ ] Feature
- [ ] Code style update (formatting, renaming)
- [ ] Refactoring (no functional changes, no api changes)
- [ ] Build related changes
- [ ] Documentation content changes
- [x] Testing
- [ ] Other (please describe): 
## What is the current behavior?

- 
## What is the new behavior?

- Add testing for components and reducers
## Does this introduce a breaking change?
- [ ] Yes
- [x] No

## Other information

- Did not add tests for setStudentId reducer.  There is a note in the file that initial state should be an object so I didn't want to change this without knowing what it should look like.
Screen Shot 2020-09-13 at 11 36 48 AM

### CODE REVIEW

File: src/CreateStudentForm/CreateStudentForm.test.js [hunk 1]
```diff
@@ -0,0 +1,87 @@
...
+import '@testing-library/jest-dom/extend-expect';
+import { render, fireEvent } from '@testing-library/react';
+import CreateStudentForm from './CreateStudentForm';
+import { Provider } from 'react-redux';
+import { createStore } from 'redux'
+import { rootReducer } from '../reducers';
+
+const store = createStore(rootReducer, {
+
+})
```
Comment (Billwilke42) [inline src/CreateStudentForm/CreateStudentForm.test.js:11]: Great work on these tests. I know I need to add some more.

File: src/reducers/setLessons.test.js [hunk 2]
```diff
@@ -1,20 +1,30 @@
 import { setLessons } from './setLessons'
-import { lesson } from '../mockData/mockData' 
+import lesson from '../mockData/mockData'
+
+
+const lessons = [lesson, lesson]
 
 describe('setLessons', () => {
-  it('Should return an initial state', () => {
+  it('should return initial state', () => {
```
Comment (Billwilke42) [inline src/reducers/setLessons.test.js:8]: Yours look a lot better then the ones I wrote, nice job

File: src/reducers/setStudentId.test.js [hunk 3]
```diff
@@ -0,0 +1,7 @@
+import { setStudentId } from './setStudentId'
+
+describe('setStudentId', () => {
+  it('should return initial state', () => {
```
Comment (Billwilke42) [inline src/reducers/setStudentId.test.js:4]: we should expand on this

**Review by Billwilke42 — APPROVED** [review]
nice work, we should look at implementing propTypes on our components soon.

File: package.json [hunk 4]
```diff
@@ -43,6 +43,7 @@
   "devDependencies": {
     "@testing-library/dom": "^7.24.1",
     "@testing-library/jest-dom": "^5.11.4",
-    "@testing-library/react": "^11.0.2"
+    "@testing-library/react": "^11.0.2",
+    "jest-environment-jsdom-sixteen": "^1.0.3"
   }
```
Comment (Melizzo) [inline package.json:48]: That patch!!! jsdom-sixteen!

File: src/Modal/Modal.js [hunk 5]
```diff
@@ -5,7 +5,7 @@ const Modal = ({ content, toggleDisplay }) => {
   return (
     <section className='modal'>
       <button onClick={toggleDisplay}>X</button>
-      {content}
+      {content || <p data-testid='no-content'>Sorry, no content at this time.  Please try again.</p>}
     </section>
   )
```
Comment (Melizzo) [inline src/Modal/Modal.js:10]: Nice - I always forget about the loading of content

File: src/TeacherRoster/TeacherRoster.js [hunk 6]
```diff
@@ -13,7 +13,7 @@ const TeacherRoster = ({ students }) => {
   const renderStudentNames = () => {
     return students.map(student => {
       return (
-        <p id={student.id} onClick={(e) => findStudent(e)}>{student.first_name} {student.last_name}</p>
+        <p id={student.id} key={student.id} onClick={(e) => findStudent(e)}>{student.first_name} {student.last_name}</p>
       )
```
Comment (Melizzo) [inline src/TeacherRoster/TeacherRoster.js:17]: This was really smart thinking - on click of the student, grab the id display the name.

**Review by Melizzo — APPROVED** [review]

File: src/reducers/setStudentId.test.js (hunk 3 above)
Comment (KyleWong2510) [reply src/reducers/setStudentId.test.js:4]: Yea I wasn't sure what this needed to actually be so I left this one for now.

**Pull request merged**